server/gateway.go

```diff
@@ -2444,7 +2447,25 @@ func (c *client) sendMsgToGateways(acc *Account, msg, subject, reply []byte, qgr
 			mh = append(mh, mreply...)
 			mh = append(mh, ' ')
 		}
-		mh = append(mh, c.pa.szb...)
+		// Headers
+		hasHeader := c.pa.hdr > 0
+		canReceiveHeader := gwc.headers
```

Comment:
I understand that `.headers` is a boolean that is immutable and we should not get a race since we set this value for an outbound before registering it (that is, before becoming available in this function).
If it becomes an issue, I would move this down into `if hasHeader` and grab the gwc's lock.

Reply:
Have not seen any issues but agree.